.teamcity/_self/projects/WebApp.kt

```diff
@@ -571,23 +571,67 @@ object CheckCodeStyleBranch : BuildType({
 			name = "Run eslint"
 			scriptContent = """
 				set -x
+
 				export NODE_ENV="test"
 
-				# Find files to lint
-				TOTAL_FILES_TO_LINT=$(git diff --name-only --diff-filter=d refs/remotes/origin/trunk...HEAD | grep -cE '(\.[jt]sx?|\.json|\.md)${'$'}' || true)
+				# Find lintable files touched in this branch, except those deleted
+				function _find_files_to_lint() {
+					git diff --name-only --diff-filter=d refs/remotes/origin/trunk...HEAD \
+						| grep -E '(\.[jt]sx?|\.json|\.md)${'$'}' \
+						|| true
+				}
+
+				# Use with `grep -c .` to prevent miscounts due to newlines
+				FILE_COUNT=${'$'}(_find_files_to_lint | grep -c . || true)
 
-				# Avoid running more than 16 parallel eslint tasks as it could OOM
-				if [ "%run_full_eslint%" = "true" ] || [ "${'$'}TOTAL_FILES_TO_LINT" -gt 16 ] || [ "${'$'}TOTAL_FILES_TO_LINT" == "0" ]; then
+				# Create temporary output directory. Export the variable so that it is
+				# available in the batch runs.
+				export RESULTS_DIR=checkstyle_results/eslint
+				mkdir -p "${'$'}RESULTS_DIR"
+
+				if [ "%run_full_eslint%" = true ] || [ "${'$'}FILE_COUNT" -eq 0 ]; then
 					echo "Linting all files"
-					yarn run eslint --format checkstyle --output-file "./checkstyle_results/eslint/results.xml" .
+					yarn run eslint --format checkstyle --output-file "${'$'}RESULTS_DIR/results.xml" .
 				else
-					# To avoid `ENAMETOOLONG` errors linting files, we have to lint them one by one,
-					# instead of passing the full list of files to eslint directly.
-					for file in ${'$'}(git diff --name-only --diff-filter=d refs/remotes/origin/trunk...HEAD | grep -E '(\.[jt]sx?|\.json|\.md)${'$'}' || true); do
-						( echo "Linting ${'$'}file"
-						yarn run eslint --format checkstyle --output-file "./checkstyle_results/eslint/${'$'}{file//\//_}.xml" "${'$'}file" ) &
-					done
-					wait
+					echo "Linting affected files"
+
+					# In an ideal scenario, we'd simply pipe the list of target files to `xargs`:
```

Comment:
It'd be great to add a comment explaining _why_ we batch (to prevent both OOM and ENAMETOOLONG errors). That's a piece of knowledge that future readers (including ourselves) will only get if they dig into git history and find this PR.

Reply:
You're right; I'll put it back.